### PR TITLE
add blockaid scan for custom mainnet and use blockaid lib for types

### PR DESCRIPTION
**Pull request by piyalbasu**
We should make sure that users who are using Mainnet with a custom network (a common occurrence with people wanting to use their rpc/horizon instances) still get the benefit of Blockaid asset scanning

### CODE REVIEW

File: @shared/api/package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "version": "1.0.0",
   "private": true,
   "dependencies": {
+    "@blockaid/client": "^0.25.0",
```
Comment (piyalbasu) [inline @shared/api/package.json:7]: I'm adding the lib as a dep strictly for the types. I thought it'd be nice to not have to worry about updating types with subsequent releases. The cost here is the added bundle size, but imo not a major concern with an extension

File: @shared/api/types.ts [hunk 2]
```diff
@@ -245,10 +246,9 @@ export interface Balance {
   blockaidData: BlockAidScanAssetResult;
 }
 
-export interface BlockAidScanAssetResult {
-  result_type: "Benign" | "Warning" | "Malicious" | "Spam";
-  features: { description: string }[];
-}
+export type BlockAidScanAssetResult = Blockaid.TokenScanResponse;
+export type BlockAidScanSiteResult = Blockaid.SiteScanResponse;
+export type BlockAidScanTxResult = Blockaid.StellarTransactionScanResponse;
```
Comment (piyalbasu) [inline @shared/api/types.ts:251]: I'm exporting all the types from this file so we only have to worry about keeping "@blockaid/client" updated in one subrepo

File: extension/src/popup/views/GrantAccess/index.tsx [hunk 3]
```diff
@@ -48,6 +48,9 @@ export const GrantAccess = () => {
     window.close();
   };
 
+  const isMalicious =
+    data && "is_malicious" in data ? data.is_malicious : false;
```
Comment (piyalbasu) [inline extension/src/popup/views/GrantAccess/index.tsx:52]: was getting a type error with the new type as "is_malicious" doesn't exist when Blockaid returns a miss

File: extension/src/popup/views/__tests__/Account.test.tsx [hunk 4]
```diff
@@ -168,6 +256,85 @@ describe("Account view", () => {
       expect(screen.getAllByText("USDC")).toBeDefined();
     });
   });
+  it("displays balances and scam notifications on custom Mainnet network", async () => {
```
Comment (piyalbasu) [inline extension/src/popup/views/__tests__/Account.test.tsx:259]: these tests also give us a little cover on the custom network experience, which we sorely need

Comment (aristidesstaffieri) [reply extension/src/popup/views/__tests__/Account.test.tsx:259]: nice, thanks

**Pull request merged**